src/test/codegen/repeat-trusted-len.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -O
+
+#![crate_type = "lib"]
+
+use std::iter;
+
+// CHECK-LABEL: @repeat_take_collect
+#[no_mangle]
+pub fn repeat_take_collect() -> Vec<u8> {
+// CHECK: call void @llvm.memset.p0i8
+    iter::repeat(42).take(100000).collect()
+}
+
+// CHECK-LABEL: @range_from_take_collect
+#[no_mangle]
+pub fn range_from_take_collect() -> Vec<u8> {
+// CHECK: %broadcast.splatinsert = insertelement <{{[0-9]+}} x i8> undef, i8 %{{.*}}, i32 0
+// CHECK: %broadcast.splat = shufflevector <[[WIDTH:[0-9]+]] x i8> %broadcast.splatinsert, <[[WIDTH]] x i8> undef, <[[WIDTH]] x i32> zeroinitializer
```

Comment:
Should I use `CHECK-NEXT` here instead?

Reply:
I don't know the details here. Using CHECK-NEXT where possible seems very good. Are these the right instructions to match on? If it's too fragile or not meaningfully testing codegen it might be better to skip this function.

Reply:
This test might be a bit fragile, but I don't know how else I'd test for llvm vectorizing the operations. In this test I'm checking that llvm extends the last written number into a vector, which should indicate with high certainty that it'll perform vectorized operations.